woocommerce/woocommerce-gutenberg-products-block
# Dynamic grid blocks: Try using ServerSideRender component for preview

See https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/558#discussion_r283286010 – In that PR, the blocks are dynamic - rendered in PHP but previewed with a JS component. It would be nice if we could keep all the rendering in one place. With the `ServerSideRender` component, we can call an API endpoint which returns the `render_callback` content given the attributes passed in.

This also means we can remove the code for fetching products from the API endpoint, and let the PHP callback do all the work.

I've marked this as in progress, but would like some feedback on the approach before I keep going.

#### Screenshots

Currently this makes for an ugly preview, but it _is_ pulling from the PHP function. We might need to add CSS to the editor, or tweak the HTML in the PHP function… either way this needs work.

<img width="706" alt="Screen Shot 2019-05-14 at 5 36 50 PM" src="https://user-images.githubusercontent.com/541093/57734609-3ccd1a00-7670-11e9-8cc5-293ff85f8566.png">

#### Accessibility

Note, this PR also wraps the content in `Disabled`, which prevents the links/buttons from being clickable. Before merging, we'll want to run through and see how this behaves with a screen reader.

### How to test the changes in this Pull Request:

1. Add a Newest Products or On Sale Products block
2. It should preview correctly, with the expected products/order/etc
3. Publish your post
4. View it, it should have the same (correct) products in the same order.
